### PR TITLE
Implement advanced player statuses indicator

### PR DESCRIPTION
**Pull request by Metadorius**
This pull request implements player slot indicators which show more information than simply ready or not ready.

example

Supported states are:
- Empty
- Unavailable
- AI
- Not Ready,
- Ready
- In Game
- Warning
- Error

### CODE REVIEW

File: DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs [hunk 1]
```diff
@@ -116,6 +121,8 @@ public override void Initialize()
 
             base.Initialize();
 
+            // DisableSpectatorReadyChecking = GameOptionsIni.GetBooleanValue("General", "DisableSpectatorReadyChecking", false);
+
```
Comment (devo1929) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs:125]: do these commented lines and the ones on 81 need to stay?

File: DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs [hunk 2]
```diff
@@ -840,19 +851,38 @@ protected override void BtnLaunchGame_LeftClick(object sender, EventArgs e)
...
+                }
+                else
+                {
+                    if (!player.Ready)
                     {
                         GetReadyNotification();
+                        return;
                     }
+                }
+                */
```
Comment (devo1929) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs:878]: more commented lines. do they need to stay in?

**Comment by devo1929** [comment]
Everything seems to look good to me. There's a lot of commented out code that can probably just be removed.

There are conflicts with the latest Rampa libraries.

Do we need to theme the icons for the various YR themes? (I'll offer to do so, if the answer is yes).

File: ClientGUI/XNAPlayerSlotIndicator.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                    ToolTip.Text = "The player is in game.";
+                    break;
+
+                case PlayerSlotState.Warning:
+                    ToolTip.Text = "The player has some issue(s) that may impact gameplay.";
+                    break;
+
+                case PlayerSlotState.Error:
+                    ToolTip.Text = "There's a critical issue with the player.";
+                    break;
```
Comment (SadPencil) [inline ClientGUI/XNAPlayerSlotIndicator.cs:89]:
```suggestion
            switch (key)
            {
                case PlayerSlotState.Empty:
                    ToolTip.Text = "The slot is empty.".L10N("UI:ClientGUI:SlotEmpty");
                    break;

                case PlayerSlotState.Unavailable:
                    ToolTip.Text = "The slot is unavailable.".L10N("UI:ClientGUI:SlotUnavailable");
                    break;

                case PlayerSlotState.AI:
                    ToolTip.Text = "The player is computer-controlled.".L10N("UI:ClientGUI:PlayerIsComputer");
                    break;

                case PlayerSlotState.NotReady:
                    ToolTip.Text = "The player isn't ready.".L10N("UI:ClientGUI:PlayerIsNotReady");
                    break;

                case PlayerSlotState.Ready:
                    ToolTip.Text = "The player is ready.".L10N("UI:ClientGUI:PlayerIsReady");
                    break;

                case PlayerSlotState.InGame:
                    ToolTip.Text = "The player is in game.".L10N("UI:ClientGUI:PlayerIsInGame");
                    break;

                case PlayerSlotState.Warning:
                    ToolTip.Text = "The player has some issue(s) that may impact gameplay.".L10N("UI:ClientGUI:PlayerHasIssue");
                    break;

                case PlayerSlotState.Error:
                    ToolTip.Text = "There's a critical issue with the player.".L10N("UI:ClientGUI:PlayerHasCriticalIssue");
                    break;
```

File: ClientGUI/XNAPlayerSlotIndicator.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using Rampastring.XNAUI;
+using Rampastring.XNAUI.XNAControls;
+using System;
+using System.Collections.Generic;
+
```
Comment (SadPencil) [inline ClientGUI/XNAPlayerSlotIndicator.cs:6]:
```suggestion
using Localization;

```

File: DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs [hunk 5]
```diff
@@ -267,6 +269,11 @@ protected override void StartGame()
             if (fsw != null)
                 fsw.EnableRaisingEvents = true;
 
+            for (int pId = 0; pId < Players.Count; pId++)
+            {
+                Players[pId].IsInGame = true;
+            }
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs:275]:
```suggestion
                Players[pId].IsInGame = true;
```

**Comment by Metadorius** [comment]
Waiting for TSC config as the default one and fixed develop branch.

**Pull request merged**